### PR TITLE
osmodel.mak: add comments

### PR DESCRIPTION
**Pull request by WalterBright**
Makefiles need comments, too.

### CODE REVIEW

File: osmodel.mak [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+#
+# Detects and sets the macros:
+#
+#   OS         = one of {osx,linux,freebsd,openbsd,netbsd,solaris}
+#   MODEL      = one of { 32, 64 }
+#   MODEL_FLAG = one of { -m32, -m64 }
+#
+# Note:
+#   Keep this file in sync between druntime, phobos, and dmd repositories!
+# Source: https://github.com/dlang/druntime/blob/master/osmodel.mak
```
Comment (CyberShadow) [inline osmodel.mak]: I don't understand the purpose of this source link.

File: osmodel.mak [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+#   osmodel.mak
+#
+# Detects and sets the macros:
+#
+#   OS         = one of {osx,linux,freebsd,openbsd,netbsd,solaris}
+#   MODEL      = one of { 32, 64 }
+#   MODEL_FLAG = one of { -m32, -m64 }
+#
+# Note:
+#   Keep this file in sync between druntime, phobos, and dmd repositories!
```
Comment (CyberShadow) [inline osmodel.mak]: I guess this needs accompanying PRs to dmd and phobos as per the comment.

File: osmodel.mak (hunk 1 above)
Comment (WalterBright) [inline osmodel.mak]: We have them in all the dmd source code files - it's for the same reason - an easy way to find out things like the history of the file.

File: osmodel.mak (hunk 2 above)
Comment (WalterBright) [inline osmodel.mak]: Yes, but after this one.

**Comment by andralex** [comment]
codecov should not consider this for coverage. Why is jenkins failing? cc @MartinNowak

**Comment by ZombineDev** [comment]
https://ci.dlang.io/blue/organizations/jenkins/dlang-org%2Fdruntime/detail/PR-1843/1/pipeline#step-480-log-330 - looks like configuration / environment issue. We can try restarting the Jenkins by closing & reopening the PR.

**Comment by andralex** [comment]
let's see

File: osmodel.mak (hunk 2 above)
Comment (wilzbach) [inline osmodel.mak]: > I guess this needs accompanying PRs to dmd and phobos as per the comment.

FWIW Phobos and Druntime already depend on DMD and I see no logical reason to have redundant copies of this files...
I actually tried to push for this once:

https://github.com/dlang/phobos/pull/5228

But was stopped by people not wanting to have the DMD Makefile auto-checking out the tools repo.
However, we could go with the small scale solution and at least get rid of the osmodel files at Phobos, Druntime ...
Dlang.org was controversial as well due to automatic git clone

Comment (andralex) [inline osmodel.mak]: I wonder if there's a way to fix this from the github side - is it possible to link a file across projects?

Comment (CyberShadow) [inline osmodel.mak]: > I wonder if there's a way to fix this from the github side - is it possible to link a file across projects?

Yes, it's called submodules, and they come with their share of problems. (You need to remember to do `git submodule update` any time you move HEAD.)

File: osmodel.mak (hunk 1 above)
Comment (wilzbach) [inline osmodel.mak]: `git log osmodel.mak`, no?

**Pull request merged**